### PR TITLE
Fix leaking goroutines in kubelet integration test

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/108483

```release-note
NONE
```

/kind cleanup
/priority important-longterm
/sig node

### CODE REVIEW

File: pkg/kubelet/util/manager/watch_based_manager.go [hunk 1]
```diff
@@ -295,7 +301,10 @@ func (c *objectCache) Get(namespace, name string) (runtime.Object, error) {
 	// This protects from premature (racy) reflector closure.
 	item.setLastAccessTime(c.clock.Now())
 
-	item.restartReflectorIfNeeded()
+	// Don't restart reflector if Kubelet is already shutting down.
+	if !c.stopped {
```
[the remark this answers is not in the document]
Comment (aojea) [reply pkg/kubelet/util/manager/watch_based_manager.go:305]: by the wayu, should not be locked the whole Get operation ? 
it gets an item under the lock, then it releases the lock and keep working on the item ... but it seems that you can Delete the item in parallel

[the remark this answers is not in the document]
Comment (wojtek-t) [reply pkg/kubelet/util/manager/watch_based_manager.go:305]: Good catch about the addition.

Regarding being blocked for the whole Get() operation - it's not needed.
The lock is only locking the "map" as a whole. Individual items are being locked at the "item" level (each item has its own lock):
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/util/manager/watch_based_manager.go#L58

**Comment by wojtek-t** [comment]
@aojea - thanks, PTAL

**Comment by wojtek-t** [comment]
Unrelated flake:
/pkg/controller/volume/attachdetach/reconciler: Test_Run_OneVolumeDetachFailNodeWithReadWriteOnce

/retest

**Pull request merged**